packages/runtime/container-runtime-definitions/src/snapshot.ts

```diff
@@ -0,0 +1,53 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISnapshotTree } from "@fluidframework/protocol-definitions";
+
+export const missingSnapshotFormatVersion = undefined;
+
+export const nextContainerSnapshotFormatVersion = "1.0";
+export type ContainerRuntimeSnapshotFormatVersion =
+    | typeof missingSnapshotFormatVersion
+    | typeof nextContainerSnapshotFormatVersion;
+
+export const currentDataStoreSnapshotFormatVersion = "1.0";
+export const nextDataStoreSnapshotFormatVersion = "2.0";
+export type DataStoreSnapshotFormatVersion =
+    | typeof missingSnapshotFormatVersion
+    | typeof currentDataStoreSnapshotFormatVersion
+    | typeof nextDataStoreSnapshotFormatVersion;
+
+export const metadataBlobName = ".metadata";
+export const chunksBlobName = ".chunks";
+export const blobsTreeName = ".blobs";
+
+export interface IContainerRuntimeMetadata {
+    snapshotFormatVersion: ContainerRuntimeSnapshotFormatVersion;
+}
+
+const protocolTreeName = ".protocol";
+
+/**
+ * List of tree IDs at the container level which are reserved.
+ * This is for older versions of snapshots that do not yet have an
+ * isolated data stores namespace. Without the namespace, this must
+ * be used to prevent name collisions with data store IDs.
+ */
+export const nonDataStorePaths = [protocolTreeName, ".logTail", ".serviceProtocol", blobsTreeName];
+
+export const dataStoreAttributesBlobName = ".component";
+
+export interface IRuntimeSnapshot {
```

Comment:
I was thinking to more strongly type the other layers as well (and actually wrote out the types as seen in the first commit), but it requires more work across the boundaries with not much gain right now.  Will look more into it after fixing SummarizerNode.

Reply:
I think id (at that level) should be always null., no?
Based on latest discussion with SPO, I think we should start asserting in our layers that we provide either value (content but id === null) or reference (id !== null, but no other fields are provided) for trees, similar how we do for blobs - we either reuse existing one, or write out new one.

Reply:
My understanding was that storage always returns ID + all tree contents (i.e. this is the full skeleton). I think it could make sense for them to sometimes return partial trees, but I don't know exactly how that would be specified right now.

If there's a better way to type it that you know of, I can change it, but I know just from debugging that currently it will be id !== null _and_ other fields are provided so that type would be misleading or at least limiting (maybe intentionally?).

Reply:
Sorry, I though this interface is for writing snapshots (summaries), not for reading.
Runtime never reads "snapshots", it operates in individual (shallow) trees that always have content and ID.

BTW, we should be try to eliminate "snapshot" word from our code base. It is used as in "snapshot tree but I think naming is from the past. I think they are always shallow trees (i.e. only contain one level of data), right?